src/server_manager/web_app/cloud_accounts.ts

```diff
@@ -12,28 +12,129 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as cloud from '../model/cloud';
 import * as digitalocean from '../model/digitalocean';
+import * as gcp from '../model/gcp';
+import {DigitalOceanAccount, ShadowboxSettings} from './digitalocean_account';
+import {GcpAccount} from './gcp_account';
 
-type DigitalOceanAccountFactory = (accessToken: string) => digitalocean.Account;
+export type AccountJson = {
+  digitalocean?: DigitalOceanAccountJson,
+  gcp?: GcpAccountJson,
+};
 
-export class CloudAccounts {
-  private readonly DIGITALOCEAN_TOKEN_STORAGE_KEY = 'LastDOToken';
+type DigitalOceanAccountJson = {
+  accessToken: string
+};
+
+type GcpAccountJson = {
+  refreshToken: string,
+};
+
+/**
+ * Manages connected cloud provider accounts.
+ */
+export class CloudAccounts implements cloud.CloudAccounts {
+  private readonly LEGACY_DIGITALOCEAN_STORAGE_KEY = 'LastDOToken';
+  private readonly ACCOUNTS_STORAGE_KEY = 'accounts-storage';
```

Comment:
Perhaps just `accounts`? Or `cloud-accounts`? `storage` is implied

Reply:
Replaced with `accounts.digitalocean` and `accounts.gcp`